### PR TITLE
Added #3957 : Label generation and bulk actions for Consumables 

### LINKED ISSUE #3957: [Feature Request] Barcode generation for Licenses, Accessories, Consumables, and Components
#### Expected Behavior (or desired behavior if a feature request)

Have the ability to generate a barcode for Licenses, Accessories, Consumables, and Components just like you can for assets.

The desire is to be able to use this for storage room inventory. I would like to be able to put my keyboards, mice, phones, laptops, ect... on storage shelves and have a barcode (2D QRCode) that I can scan and will take me right to the page where I can check out the item (even better would be just remove one from the qty and have the option to not have to assign it to someone)

Right now you can only do this with Asset items, but you can't say you have 30 of the same item.

I did try to use a QRcode generator with a URL such as 
> http://snipeit/admin/accessories/1/view
But that just ended up taking me to a login page and then after logging in, to the dashboard.
-----

#### Actual Behavior

There appears to be no barcode generation available for anything other than assets.

-----

#### Please confirm you have done the following before posting your bug report:

- [ ] I have enabled debug mode
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running: Version v3.6.6 build 1 (gee2c67a)
- What OS and web server you're running Snipe-IT on: Centos 6.9 Apache
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) Manual installation. Install.sh appears broken.
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error: no Errors
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way: only got errors with the install.sh saying it couldn't complete the install because of the OS version. I just restarted and manually installed. There were no errors when manually installing.
- Indicate whether or not you've manually edited any data directly in the database: no custom editing.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/49307661-feature-request-barcode-generation-for-licenses-accessories-consumables-and-components?utm_campaign=plugin&utm_content=tracker%2F505106&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F505106&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by 01ste02**
# Description
You can now perform bulk actions on Consumables, as well as generate labels for them.
image
image
image
image

Fixes #3957 

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I ran the old tests, and tested the bulk actions and label generation manually. The qr-codes point to the correct location. However, the barcodes point to "id-"and then the consumable id. This prefix is needed because I use the consumable id which is sometimes too short for the barcode generator to accept. The prefix could be used to differentiate between consumables and hardware in the topmost searchbar (for barcode scanners).

**Test Configuration**:
* PHP version: 7.4.20
* MySQL version: Ver 15.1 Distrib 10.5.11-MariaDB
* Webserver version: httpd-2.4.48
* OS version: Fedora release 34 (Thirty Four)

# Checklist:

- [X] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [X] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by smcpeck** [comment]
@snipe @01ste02 While this PR is a bit old, I think it would be a great additional still.

Eager to try and help if I can do anything to move this forward.  Would be willing to take this into my own fork and run it through the wringer a bit...

My concerns are:
1. While I am broadly capable, I am not a PHP/Laravel guy.
2. Being that this is a bit old, I suspect it has diverged a good amount. Maybe you can sync things back up @01ste02?

Let me know if there is some specific process I should review/follow. Happy to assist, just need a little guidance.

File: app/Http/Controllers/Consumables/BulkConsumablesController.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    public function destroy(Request $request)
+    {
+        $this->authorize('delete', Consumable::class);
+
+        if ($request->filled('ids')) {
+            $consumables = Consumable::find($request->get('ids'));
+            foreach ($consumables as $consumable) {
+                $update_array['deleted_at'] = date('Y-m-d H:i:s');
+
+                DB::table('consumables')
```
Comment (snipe) [inline app/Http/Controllers/Consumables/BulkConsumablesController.php:186]: I think we can just use the `->delete()` method here?

Also how are we accounting for consumables that have been checked out already?

**Comment by 01ste02** [comment]
I sadly do not have the time to sync this request currently, but suspect that it might not be all that difficult to sync up. I might have the time in a couple of months, but as of now I cannot fix this.

**Comment by snipe** [comment]
Closing this for now, just for housekeeping - feel free to re-open though!

**Comment by Jsandy78** [comment]
Any way this can still be implemented? Could really use this feature currently for checking out toners to specific people among other things?

**Pull request closed without merging**